sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/circuitBreaker/ConsecutiveExceptionBasedCircuitBreaker.java

```diff
@@ -0,0 +1,219 @@
+                        .build();
+
+                } else {
+
+                    return new LocationSpecificHealthContext.Builder()
+                        .withSuccessCountForWriteForRecovery(locationSpecificHealthContext.getSuccessCountForWriteForRecovery())
+                        .withExceptionCountForWriteForCircuitBreaking(exceptionCountAfterHandling)
+                        .withSuccessCountForReadForRecovery(locationSpecificHealthContext.getSuccessCountForReadForRecovery())
+                        .withExceptionCountForReadForCircuitBreaking(locationSpecificHealthContext.getExceptionCountForReadForCircuitBreaking())
+                        .withUnavailableSince(locationSpecificHealthContext.getUnavailableSince())
+                        .withLocationHealthStatus(locationSpecificHealthContext.getLocationHealthStatus())
+                        .withExceptionThresholdBreached(locationSpecificHealthContext.isExceptionThresholdBreached())
+                        .build();
+                }
+            case HealthyTentative:
+
+                successCountAfterHandling++;
+
+                if (isReadOnlyRequest) {
+
+                    return new LocationSpecificHealthContext.Builder()
+                        .withSuccessCountForWriteForRecovery(locationSpecificHealthContext.getSuccessCountForWriteForRecovery())
+                        .withExceptionCountForWriteForCircuitBreaking(locationSpecificHealthContext.getExceptionCountForWriteForCircuitBreaking())
+                        .withSuccessCountForReadForRecovery(successCountAfterHandling)
+                        .withExceptionCountForReadForCircuitBreaking(exceptionCountAfterHandling)
+                        .withUnavailableSince(locationSpecificHealthContext.getUnavailableSince())
+                        .withLocationHealthStatus(locationSpecificHealthContext.getLocationHealthStatus())
+                        .withExceptionThresholdBreached(locationSpecificHealthContext.isExceptionThresholdBreached())
+                        .build();
+
+                } else {
+
+                    return new LocationSpecificHealthContext.Builder()
+                        .withSuccessCountForWriteForRecovery(successCountAfterHandling)
+                        .withExceptionCountForWriteForCircuitBreaking(exceptionCountAfterHandling)
+                        .withSuccessCountForReadForRecovery(locationSpecificHealthContext.getSuccessCountForReadForRecovery())
+                        .withExceptionCountForReadForCircuitBreaking(locationSpecificHealthContext.getExceptionCountForReadForCircuitBreaking())
+                        .withUnavailableSince(locationSpecificHealthContext.getUnavailableSince())
+                        .withLocationHealthStatus(locationSpecificHealthContext.getLocationHealthStatus())
+                        .withExceptionThresholdBreached(locationSpecificHealthContext.isExceptionThresholdBreached())
```

Comment:
a lot of these calls are the same in both clauses - you can perhaps make the common calls first and then have if clause with only the few changes and a return a build call after. Would make it easier to see what is the actual differenece in the clauses.

Reply:
Makes sense - will split the with* calls.

Reply:
Fixed in next iteration.